rocksdb_replicator/rocksdb_replicator.h

```diff
@@ -136,7 +136,9 @@ class RocksDBReplicator {
     std::shared_ptr<replicator::DbWrapper> db_wrapper_;
     folly::Executor* const executor_;
     const DBRole role_;
+    const ReplicaRole role_thrift_;
```

Comment:
Why do we need two *roles*? Won't DBRole is actually the replication role?

Reply:
This is an optimization: we shouldn't need to pass string over the wire on every replication response which can be expensive

Reply:
`DBRole` is *enum*, right? not a string: https://github.com/pinterest/rocksplicator/blob/master/rocksdb_replicator/rocksdb_replicator.h#L69

Reply:
Sorry, should have made it clearer: we cannot pass `DBRole` directly into the thrift replication response, so we would need to define a thrift enum instead of the `DBRole` enum as you pointed above. 

I thought about using the thrift enum everywhere and get rid of the existing `DBRole` enum, but that will cause a large blast radius of refactor: https://github.com/pinterest/rocksplicator/search?p=3&q=DBRole. Something I can follow up. 

There is also another difference I forgot to callout: `DBRole` has `NOOP` as well which is a validate replication role (e.g. in [BOOTSTRAP](https://github.com/pinterest/rocksplicator/blob/63c9c7d738dc868e59c552a43e2b305314092b0c/cluster_management/src/main/java/com/pinterest/rocksplicator/BootstrapStateModelFactory.java#L129) mode I think), but for the replication flow, only LEADER/FOLLOWER is concerned, so these two enums are not exactly the same.

Reply:
I think we need unify them, it will cause confusion. 

There is really only 1 role for the db, which is defined by *Replication*. 
The *Noop* just tell the replicationDB that it does not replicate from anyone. https://github.com/pinterest/rocksplicator/blob/63c9c7d738dc868e59c552a43e2b305314092b0c/rocksdb_admin%2Fadmin_handler.cpp#L616-L617

Reply:
On top of the cpp, at the cluster management layer, we have another *role*, which Helix partition role, such as Helix partition Leader, Folllower, etct, which defined by statemodel.

Reply:
Should we consider other names than `role_thrift_`? Maybe, `replica_role_` if the class is `ReplicaRole`

Reply:
Makes sense, In that case, does my above proposal works for you?
> I thought about using the thrift enum everywhere and get rid of the existing DBRole enum, but that will cause a large blast radius of refactor: https://github.com/pinterest/rocksplicator/search?p=3&q=DBRole. Something I can follow up.

Or do you prefer we do that first before this change. I am okay either way

Reply:
I will try put up a PR anyway

Reply:
See https://github.com/pinterest/rocksplicator/pull/592, can you review that first